build

```diff
@@ -64,9 +64,41 @@ function progress() {
 }
 
 TAG=${TAG:-latest}
+TAG_FILE=${TAG_FILE:-}
 CONTEXT=${CONTEXT:-.}
 DOCKERFILE=${DOCKERFILE:-$CONTEXT/Dockerfile}
 BUILD_ARGS_OPT=$(env | awk '/BUILD_ARG_/ {gsub(/BUILD_ARG_/, "--build-arg "); printf "%s ",$0}')
+BUILD_ARGS_FILE=${BUILD_ARGS_FILE:-}
+
+tag_name=""
+if [ -n "$TAG_FILE" ]; then
+  if [ ! -f "$TAG_FILE" ]; then
+    echo "tag file '$TAG_FILE' does not exist"
+    exit 1
+  fi
+  tag_name="$(cat $TAG_FILE)"
+else
+  tag_name="$TAG"
+fi
+
+build_args=""
+if [ -n "$BUILD_ARGS_FILE" ]; then
+  if [ ! -f "$BUILD_ARGS_FILE" ]; then
+    echo "build_args file '$BUILD_ARGS_FILE' does not exist"
+    exit 1
+  fi
+
+  expanded_build_args=()
+  while IFS= read -r line || [ -n "$line" ];
```

Comment:
Should this be `&&`? What's this fixing?

Reply:
The problem is without `|| [ -n "$line" ]`, shell script read will missing last line
Ref:
* https://stackoverflow.com/a/12919766/4015691
* https://stackoverflow.com/questions/12916352/shell-script-read-missing-last-line

Reply:
Ahh, ok, because `read` returns `1` when it reaches end-of-file without a final linebreak, so it checks if there was anything read before then.